src/seer/inference_models.py

```diff
@@ -12,29 +16,98 @@ def model_path(subpath: str) -> str:
     return os.path.join(root, "models", subpath)
 
 
-@functools.cache
+_A = TypeVar("_A")
+_deferred: list[Callable[[], Any]] = []
+
+
+def deferred_loading(env_var_required: str) -> Callable[[Callable[[], _A]], Callable[[], _A]]:
+    def decorator(func: Callable[[], _A]) -> Callable[[], Any]:
+        if os.environ.get(env_var_required, "").lower() not in ("true", "t", "1"):
+            return func
+        # python functools is thread safe: https://docs.python.org/3/library/functools.html
+        wrapped = functools.cache(func)
+        _deferred.append(wrapped)
+        return wrapped
+
+    return decorator
+
+
+@deferred_loading("SEVERITY_ENABLED")
 def embeddings_model() -> SeverityInference:
     return SeverityInference(
         model_path("issue_severity_v0/embeddings"), model_path("issue_severity_v0/classifier")
     )
 
 
-@functools.cache
+@deferred_loading("GROUPING_ENABLED")
 def grouping_lookup() -> GroupingLookup:
     return GroupingLookup(
         model_path=model_path("issue_grouping_v0/embeddings"),
         data_path=model_path("issue_grouping_v0/data.pkl"),
     )
 
 
-function_env_config = {
-    "embeddings_model": "SEVERITY_ENABLED",
-    "grouping_lookup": "GROUPING_ENABLED",
-}
+LoadingResult = Literal["pending"] | Literal["loading"] | Literal["done"] | Literal["failed"]
```

Comment:
```suggestion
LoadingResult = Literal["pending", "loading", "done", "failed"]
```
why are the literals separated out?

Reply:
Oh, dude, I literally didn't know about that syntax!!!

Reply:
Oh, LOL